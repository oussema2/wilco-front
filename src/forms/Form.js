import MobxReactForm from 'mobx-react-form';
import dvr from '../plugins/dvr';

export default class Form extends MobxReactForm {
	constructor(
		{ fields = [] } = {},
		{
			hooks,
			options = {
				validateOnChange: true,
				showErrorsOnChange: false,
				validationDebounceWait: 0
			},
			plugins = dvr
		} = {}
	) {
		super( { fields }, { options, hooks, plugins } );
	}
}
