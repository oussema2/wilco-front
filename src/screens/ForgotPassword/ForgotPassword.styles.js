import { StyleSheet } from 'react-native';
import { palette } from '../../Theme';

export const styles = StyleSheet.create( {
	buttonContainer: {
		alignItems: 'center'
	},
	emailTextInput: {
		marginVertical: 24,
		marginBottom: 106
	},
	subTitle: {
		marginTop: 40,
		marginBottom: 32,
		color: palette.subtitle.default
	}
} );
