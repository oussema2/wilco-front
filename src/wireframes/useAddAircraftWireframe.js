import { useEffect, useState } from 'react';
import { makeAutoObservable } from 'mobx';
import { useNavigation } from '@react-navigation/native';
import AddAircraftPresenter from '../presenters/AddAircraftPresenter';
import CreateAircraft from '../interactors/CreateAircraft';
import EntityServiceFactory from '../services/EntityServiceFactory';
import Api from '../services/Api';
import { useRootStore } from '../providers/RootStoreProvider';
import { useSnackbarService } from '../providers/SnackbarProvider';
import { useAnalyticsService } from '../providers/AnalyticsProvider';
import { useModalService } from '../providers/ModalProvider';

const useAddAircraftWireframe = ( { previousScreen } ) => {
	const [ presenterInstance, setPresenterInstance ] = useState( null );
	const root = useRootStore();
	const navigation = useNavigation();
	const snackbarService = useSnackbarService();
	const analyticsService = useAnalyticsService();
	const modalService = useModalService();

	useEffect( () => {
		const api = new Api( { authenticationStore: root.authenticationStore } );

		const createAircraft = new CreateAircraft( {
			aircraftService: EntityServiceFactory.forAircrafts( { api } ),
			aircraftStore: root.aircraftStore,
			pilotStore: root.pilotStore
		} );

		const presenter = new AddAircraftPresenter( {
			navigation,
			createAircraft,
			previousScreen,
			snackbarService,
			analyticsService,
			modalService,
			makeAutoObservable
		} );

		setPresenterInstance( presenter );
	}, [ root ] );

	return presenterInstance;
};

export default useAddAircraftWireframe;
