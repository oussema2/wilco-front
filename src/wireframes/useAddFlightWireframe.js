import { useEffect, useState } from 'react';
import { makeAutoObservable } from 'mobx';
import { useNavigation } from '@react-navigation/native';
import { useRootStore } from '../providers/RootStoreProvider';
import { useModalService } from '../providers/ModalProvider';
import { useActionSheetService } from '../providers/ActionSheetProvider';
import { useSnackbarService } from '../providers/SnackbarProvider';
import GetCurrentPilotFromStore from '../interactors/GetCurrentPilotFromStore';
import FetchAircraftFlights from '../interactors/FetchAircraftFlights';
import EntityServiceFactory from '../services/EntityServiceFactory';
import Api from '../services/Api';
import FlightBuilder from '../builders/FlightBuilder';
import GetFlightTrack from '../interactors/GetFlightTrack';
import DeleteAircraft from '../interactors/DeleteAircraft';
import { useAnalyticsService } from '../providers/AnalyticsProvider';
import AddFlightPresenter from '../presenters/AddFlightPresenter';

const useAddFlightWireframe = ( { previousScreen } ) => {
	const [ presenterInstance, setPresenterInstance ] = useState( null );
	const rootStore = useRootStore();
	const navigation = useNavigation();
	const modalService = useModalService();
	const actionSheetService = useActionSheetService();
	const snackbarService = useSnackbarService();
	const analyticsService = useAnalyticsService();

	useEffect( () => {
		const {
			pilotStore, authenticationStore, flightStore, aircraftStore
		} = rootStore;

		const getCurrentPilotFromStore = new GetCurrentPilotFromStore( {
			store: pilotStore
		} );
		const api = new Api( { authenticationStore } );
		const flightBuilder = new FlightBuilder( {
			aircraftStore, flightStore
		} );
		const flightService = EntityServiceFactory.forFlights( {
			api, buildEntity: flightBuilder.build
		} );
		const aircraftService = EntityServiceFactory.forAircrafts( { api } );
		const fetchAircraftFlights = new FetchAircraftFlights( {
			service: flightService
		} );
		const getFlightTrack = new GetFlightTrack( {
			service: flightService
		} );
		const deleteAircraft = new DeleteAircraft( {
			service: aircraftService,
			pilotStore
		} );

		const presenter = new AddFlightPresenter( {
			getCurrentPilotFromStore,
			fetchAircraftFlights,
			getFlightTrack,
			deleteAircraft,
			navigation,
			modalService,
			actionSheetService,
			snackbarService,
			analyticsService,
			previousScreen,
			makeAutoObservable
		} );

		setPresenterInstance( presenter );
	}, [ rootStore ] );

	return presenterInstance;
};

export default useAddFlightWireframe;
