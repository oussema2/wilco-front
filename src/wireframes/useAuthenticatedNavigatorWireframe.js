import { useState } from 'react';
import { makeAutoObservable } from 'mobx';
import HelpCenterService from '../services/HelpCenterService';
import GetCurrentPilotFromStore from '../interactors/GetCurrentPilotFromStore';
import { useRootStore } from '../providers/RootStoreProvider';
import AuthenticatedNavigatorPresenter from '../presenters/AuthenticatedNavigatorPresenter';

const useAuthenticatedNavigatorWireframe = () => {
	const rootStore = useRootStore();
	const { pilotStore } = rootStore;
	const getCurrentPilotFromStore = new GetCurrentPilotFromStore( { store: pilotStore } );
	const helpCenterService = HelpCenterService.shared();
	const createPresenter = () => new AuthenticatedNavigatorPresenter( {
		getCurrentPilotFromStore,
		helpCenterService,
		makeAutoObservable
	} );

	const [ presenter ] = useState( createPresenter );

	presenter.initialize();

	return presenter;
};

export default useAuthenticatedNavigatorWireframe;
