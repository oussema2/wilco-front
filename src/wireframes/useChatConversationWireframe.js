import { useState } from 'react';
import { makeAutoObservable } from 'mobx';
import { useNavigation, useRoute } from '@react-navigation/native';
import ChatService from '../services/ChatService';
import ChatConversationPresenter from '../presenters/ChatConversationPresenter';
import { useSnackbarService } from '../providers/SnackbarProvider';
import { useAnalyticsService } from '../providers/AnalyticsProvider';

const useChatConversationWireframe = () => {
	const analyticsService = useAnalyticsService();
	const chatService = ChatService.shared();
	const navigation = useNavigation();
	const snackbarService = useSnackbarService();
	const recipientChatUser = useRoute().params?.item;
	const { uid } = useRoute().params;

	const recipientChatUid = recipientChatUser?.uid || uid;
	const pilotID = recipientChatUid.replace( /\D/g, '' );

	const createPresenter = () => new ChatConversationPresenter( {
		chatService,
		navigation,
		snackbarService,
		recipientChatUid,
		recipientChatUser,
		pilotID,
		analyticsService,
		makeAutoObservable
	} );

	const [ presenter ] = useState( createPresenter );

	return presenter;
};

export default useChatConversationWireframe;
