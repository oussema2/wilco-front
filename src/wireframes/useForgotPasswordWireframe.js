import { useState } from 'react';
import { makeAutoObservable } from 'mobx';
import { useNavigation } from '@react-navigation/native';
import { Keyboard } from 'react-native';
import { useRootStore } from '../providers/RootStoreProvider';
import { useSnackbarService } from '../providers/SnackbarProvider';
import ForgotPasswordPresenter from '../presenters/ForgotPasswordPresenter';
import SendPasswordResetEmailService from '../services/SendPasswordResetEmailService';
import SendPasswordResetEmail from '../interactors/SendPasswordResetEmail';

const useForgotPasswordWireframe = () => {
	const rootStore = useRootStore();
	const snackbarService = useSnackbarService();
	const navigation = useNavigation();
	const keyboard = Keyboard;
	const sendPasswordResetEmailService = new SendPasswordResetEmailService();
	const sendPasswordResetEmail = new SendPasswordResetEmail( { sendPasswordResetEmailService } );

	const createPresenter = () => new ForgotPasswordPresenter( {
		snackbarService,
		keyboard,
		navigation,
		rootStore,
		sendPasswordResetEmail,
		makeAutoObservable
	} );

	const [ presenter ] = useState( createPresenter );

	return presenter;
};

export default useForgotPasswordWireframe;
