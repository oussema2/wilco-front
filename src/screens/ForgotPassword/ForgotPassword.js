/* eslint-disable react/jsx-no-bind */
import React from 'react';
import { observer } from 'mobx-react-lite';
import { View } from 'react-native';
import { EMAIL } from '../../constants/formFields/forgotPasswordForm';
import { PrimaryButton } from '../../components/PrimaryButton';
import { TextInput } from '../../components/TextInput';
import HorizontalPadding from '../../components/HorizontalPadding/HorizontalPadding';
import BaseScreen from '../BaseScreen/BaseScreen';
import { styles } from './ForgotPassword.styles';
import { NavigationBar } from '../../components/NavigationBar';
import useForgotPasswordWireframe from '../../wireframes/useForgotPasswordWireframe';
import { Subtitle } from '../../components/Subtitle';

const ForgotPassword = ( ) => {
	const presenter = useForgotPasswordWireframe();

	return (
		<BaseScreen isLoading={!!presenter && presenter.isSendingEmail}>
			{!!presenter && (
				<HorizontalPadding>
					<NavigationBar
						testID="title-header"
						title={presenter.title}
						onBackArrowPress={presenter.backButtonWasPressed}
					/>

					<Subtitle
						testID="subtitle-forgotPassword-testID"
						style={styles.subTitle}
						text="Please enter your email and weâ€™ll help you set a new password."
					/>

					<TextInput
						testID="email-input"
						textInputStyle={styles.emailTextInput}
						required
						error={presenter.form.$( EMAIL ).error}
						inputProps={presenter.form.$( EMAIL ).bind()}
					/>

					<View style={styles.buttonContainer}>
						<PrimaryButton
							testID="helpMe-button"
							title={presenter.buttonTitle}
							size="small"
							onPress={presenter.form.onSubmit}
							disabled={presenter.isSubmitButtonDisabled}
						/>
					</View>
				</HorizontalPadding>
			) }
		</BaseScreen>
	);
};

ForgotPassword.propTypes = {
};

ForgotPassword.defaultProps = {
};

export default observer( ForgotPassword );
