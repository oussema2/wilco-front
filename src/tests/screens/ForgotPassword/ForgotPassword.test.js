import * as React from 'react';
import { render } from '@testing-library/react-native';
import mockUseView from '../../mocks/mockUseView';
import * as useForgotPasswordWireframe from '../../../wireframes/useForgotPasswordWireframe';
import Form from '../../../forms/Form';
import fields from '../../../forms/forgotPasswordFields';
import ForgotPassword from '../../../screens/ForgotPassword/ForgotPassword';

jest.mock( '@react-native-firebase/auth', () => ( {
	auth: jest.fn( () => {} )
} ) );

describe( 'ForgotPassword', () => {
	let screen;
	const title = 'Forgot password';
	const subTitle = 'Please enter your email and weâ€™ll help you set a new password.';
	const buttonTitle = 'Submit';
	const form = new Form( { fields } );

	const setUp = ( view = {} ) => {
		mockUseView(
			useForgotPasswordWireframe,
			{
				title,
				buttonTitle,
				form,
				...view
			} );

		screen = render( <ForgotPassword /> );
	};

	describe( 'when the app is not sending email', () => {
		beforeEach( () => {
			setUp( { isSendingEmail: false } );
		} );

		it( 'renders the ForgotPassword screen correctly', async () => {
			expect( screen.queryByTestId( 'title-header' ) ).toHaveTextContent( title );
			expect( screen.queryByTestId( 'subtitle-forgotPassword-testID' ) ).toHaveTextContent( subTitle );
			expect( screen.queryByTestId( 'helpMe-button' ) ).toHaveTextContent( buttonTitle );

			expect( screen.queryByTestId( 'email-input' ) ).not.toBeNull();

			expect( screen ).toMatchSnapshot();
		} );

		it( 'does not render the screen loader', () => {
			expect( screen.queryByTestId( 'screenLoader' ) ).toBeNull();
		} );
	} );

	describe( 'when the app is sending email', () => {
		beforeEach( () => {
			setUp( { isSendingEmail: true } );
		} );

		it( 'renders the ForgotPassword screen correctly', async () => {
			expect( screen.queryByTestId( 'title-header' ) ).toHaveTextContent( title );
			expect( screen.queryByTestId( 'subtitle-forgotPassword-testID' ) ).toHaveTextContent( subTitle );
			expect( screen.queryByTestId( 'helpMe-button' ) ).toHaveTextContent( buttonTitle );

			expect( screen.queryByTestId( 'email-input' ) ).not.toBeNull();

			expect( screen ).toMatchSnapshot();
		} );

		it( 'renders the screen loader', () => {
			expect( screen.queryByTestId( 'screenLoader' ) ).not.toBeNull();
		} );
	} );
} );
