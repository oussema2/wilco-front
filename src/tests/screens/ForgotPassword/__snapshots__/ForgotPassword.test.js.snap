// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword when the app is not sending email renders the ForgotPassword screen correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="base-screen"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
          },
          null,
        ]
      }
      testID="horizontalPadding-component"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 44,
          }
        }
        testID="title-header"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            onPress={null}
            style={
              Object {
                "alignItems": "center",
                "color": "#000000",
                "flex": 1,
                "fontFamily": "NotoSansJP-Black",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": -0.3,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
            testID="title-text"
          >
            Forgot password
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
            testID="rightPlaceholder-view"
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#5C6780",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 18,
            "fontWeight": undefined,
            "letterSpacing": undefined,
            "lineHeight": 26,
            "marginBottom": 32,
            "marginTop": 40,
            "minHeight": 22,
            "textTransform": "none",
          }
        }
        testID="subtitle-forgotPassword-testID"
      >
        Please enter your email and weâ€™ll help you set a new password.
      </Text>
      <View
        style={
          Object {
            "marginBottom": 106,
            "marginVertical": 24,
            "minHeight": 98,
          }
        }
        testID="email-input"
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#242424",
                "fontFamily": "NotoSansJP-Black",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": -0.3,
                "lineHeight": 22,
                "marginBottom": 4,
              }
            }
            testID="input-label"
          >
            Email
          </Text>
          <Text
            style={
              Object {
                "color": "#D60A50",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 14,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 4,
              }
            }
            testID="input-required"
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderBottomWidth": 0,
                  "borderColor": "#86939e",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "marginBottom": 5,
                  "minHeight": 52,
                  "paddingHorizontal": 12,
                  "textAlignVertical": "center",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={true}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="e.g. wileypost@gmail.com"
                placeholderTextColor="#A0A0A0"
                returnKeyType="default"
                scrollEnabled={true}
                secureTextEntry={false}
                selectionColor="#12A3BE"
                style={
                  Object {
                    "color": "#000000",
                    "flex": 1,
                    "fontFamily": "NotoSansJP-Bold",
                    "fontSize": 14,
                    "fontWeight": undefined,
                    "letterSpacing": 0,
                    "lineHeight": 20,
                    "minHeight": 40,
                    "paddingVertical": 0,
                  }
                }
                testID="input"
                textAlignVertical="center"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#D60A50",
                  "fontFamily": "NotoSansJP-Regular",
                  "fontSize": 12,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 16,
                  "margin": 0,
                  "marginLeft": 0,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              undefined,
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="helpMe-button"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#12A3BE",
                  "borderColor": "#2089dc",
                  "borderRadius": 26,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 52,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 164,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "NotoSansJP-Bold",
                    "fontSize": 15,
                    "fontWeight": undefined,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Submit
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</BVLinearGradient>
`;

exports[`ForgotPassword when the app is sending email renders the ForgotPassword screen correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="base-screen"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
          },
          null,
        ]
      }
      testID="horizontalPadding-component"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 44,
          }
        }
        testID="title-header"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            onPress={null}
            style={
              Object {
                "alignItems": "center",
                "color": "#000000",
                "flex": 1,
                "fontFamily": "NotoSansJP-Black",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": -0.3,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
            testID="title-text"
          >
            Forgot password
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
            testID="rightPlaceholder-view"
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#5C6780",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 18,
            "fontWeight": undefined,
            "letterSpacing": undefined,
            "lineHeight": 26,
            "marginBottom": 32,
            "marginTop": 40,
            "minHeight": 22,
            "textTransform": "none",
          }
        }
        testID="subtitle-forgotPassword-testID"
      >
        Please enter your email and weâ€™ll help you set a new password.
      </Text>
      <View
        style={
          Object {
            "marginBottom": 106,
            "marginVertical": 24,
            "minHeight": 98,
          }
        }
        testID="email-input"
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#242424",
                "fontFamily": "NotoSansJP-Black",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": -0.3,
                "lineHeight": 22,
                "marginBottom": 4,
              }
            }
            testID="input-label"
          >
            Email
          </Text>
          <Text
            style={
              Object {
                "color": "#D60A50",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 14,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "marginLeft": 4,
              }
            }
            testID="input-required"
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderBottomWidth": 0,
                  "borderColor": "#86939e",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "marginBottom": 5,
                  "minHeight": 52,
                  "paddingHorizontal": 12,
                  "textAlignVertical": "center",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={true}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onSubmitEditing={[Function]}
                placeholder="e.g. wileypost@gmail.com"
                placeholderTextColor="#A0A0A0"
                returnKeyType="default"
                scrollEnabled={true}
                secureTextEntry={false}
                selectionColor="#12A3BE"
                style={
                  Object {
                    "color": "#000000",
                    "flex": 1,
                    "fontFamily": "NotoSansJP-Bold",
                    "fontSize": 14,
                    "fontWeight": undefined,
                    "letterSpacing": 0,
                    "lineHeight": 20,
                    "minHeight": 40,
                    "paddingVertical": 0,
                  }
                }
                testID="input"
                textAlignVertical="center"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#D60A50",
                  "fontFamily": "NotoSansJP-Regular",
                  "fontSize": 12,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 16,
                  "margin": 0,
                  "marginLeft": 0,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              undefined,
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="helpMe-button"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#12A3BE",
                  "borderColor": "#2089dc",
                  "borderRadius": 26,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 52,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 164,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "NotoSansJP-Bold",
                    "fontSize": 15,
                    "fontWeight": undefined,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Submit
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.6)",
        "display": "flex",
        "flexDirection": "column-reverse",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
    testID="screenLoader"
  >
    <View
      animated={true}
      animationConfig={
        Object {
          "bounciness": 0,
        }
      }
      animationType="spring"
      indeterminate={true}
      indeterminateAnimationDuration={1000}
      onLayout={[Function]}
      progress={0}
      style={
        Array [
          Object {
            "backgroundColor": undefined,
            "borderColor": "#12A3BE",
            "borderRadius": 0,
            "borderWidth": 0,
            "overflow": "hidden",
            "width": null,
          },
          undefined,
        ]
      }
      testID="progress-bar"
      useNativeDriver={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#12A3BE",
            "height": 6,
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateX": -0,
              },
              Object {
                "scaleX": 0.30007,
              },
            ],
          }
        }
      />
    </View>
  </View>
</BVLinearGradient>
`;
